features/FEATURE_BLE/ble/services/EddystoneService.h

```diff
@@ -529,7 +529,7 @@ class EddystoneService
             DBG("attached tlmCallback every %d seconds", TlmAdvPeriod);
         }
         if (NONE == frameIndex) {
-            error("No Frames were Initialized! Please initialize a frame before starting an eddystone beacon.");
+            MBED_ERROR(MBED_MAKE_ERROR(MBED_MODULE_BLE, MBED_ERROR_CODE_BLE_NO_FRAME_INITIALIZED), "No Frames were Initialized! Please initialize a frame before starting an eddystone beacon.");
```

Comment:
I'm not sure there is value adding a new error for this specific case. 
Would it be possible to use an `INVALID_STATE` error instead ? 
We already have code such as `INVALID_ARGUMENT`.

Reply:
@pan-  The value here is to get maximum info about the error scenario just by decoding the error code even if there is no textual output. Please look at the error decoding microsite - https://armmbed.github.io/mbedos-error/. This website takes the error code and generates error context info. Once we integrate other features like device stats API for errors as well, the error-code info collected can be used to get more info. If you still think its not useful we can remove it. Let me know.

Reply:
If the file and line number is reported, is it that useful to have very specific error code ? (If file and line are not accessible; it is very useful).

Reply:
@pan- - The line number/file is not reported by default. It needs a special flag MBED_CONF_PLATFORM_ERROR_FILENAME_CAPTURE_ENABLED. Irrespective of that we plan to report these error-codes into cloud/or other backends and thus having these error codes would provide us more info for analytics/later analysis. Does that makes sense?

Reply:
That's unfortunate it requires a special flag to report the filename and line number as without it the error code in this form is not very comprehensive. 

Being dependent on a website to decipher an error code is, from my perspective, not ideal for debugging and it is also limited to the content we feed the website with. I hope their will be some way to run the webpage locally so it is in sync with the sources being run otherwise the whole error code mechanism will be very complex to use with application defined error codes.

Side question: how do you intend to capture in the cloud error codes reported when the system halt ?